contracts/contracts/eosio.system/src/voting.cpp

```diff
@@ -86,6 +86,8 @@ namespace eosiosystem {
    void system_contract::unregprod( const name& producer ) {
       require_auth( producer );
 
+      claimrewards(producer);
```

Comment:
in situation when we`ve already claimed within last day, it will fail. Is that OK?

Reply:
@me-vadim I was thinking about it; it seems not obvious but legit.